### PR TITLE
[MINOR][CORE][PYTHON][SQL][PS] Fix some typos

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Fix some typos.

### Why are the changes needed?
Make the documentation easier to read

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

**Comment by bjornjorgensen** [comment]
@srowen FYI

File: python/docs/source/conf.py [hunk 1]
```diff
@@ -224,7 +224,7 @@
 # Custom sidebar templates, maps document names to template names.
 #html_sidebars = {}
 
-# Additional templates that should be rendered to pages, maps page names to
+# Additional templates that should be rendered to pages, map page names to
```
Comment (srowen) [inline python/docs/source/conf.py:227]: I think this was correct; maybe change the comma to semicolon instead

File: python/docs/source/user_guide/pandas_on_spark/typehints.rst [hunk 2]
```diff
@@ -33,10 +33,9 @@ for schema inference, and once for processing actual data with the schema.
 
 To avoid the consequences, pandas API on Spark has its own type hinting style to specify the schema to avoid
 schema inference. Pandas API on Spark understands the type hints specified in the return type and converts it
-as a Spark schema for pandas UDFs used internally. The way of type hinting has been evolved over
-the time.
+as a Spark schema for pandas UDFs used internally. The way of type hinting has evolved over time.
 
-In this chapter, it covers the recommended way and the supported ways in details.
+In this chapter, it covers the recommended way and the supported ways in detail.
```
Comment (srowen) [inline python/docs/source/user_guide/pandas_on_spark/typehints.rst:38]: "This chapter covers ..."

File: python/docs/source/user_guide/python_packaging.rst [hunk 3]
```diff
@@ -23,7 +23,7 @@ Python Package Management
 When you want to run your PySpark application on a cluster such as YARN, Kubernetes, Mesos, etc., you need to make
 sure that your code and all used libraries are available on the executors.
 
-As an example let's say you may want to run the `Pandas UDF's examples <sql/arrow_pandas.rst#series-to-scalar>`_.
+As an example, let's say you may want to run the `Pandas UDF's examples <sql/arrow_pandas.rst#series-to-scalar>`_.
```
Comment (srowen) [inline python/docs/source/user_guide/python_packaging.rst:26]: `Pandas UDF examples` maybe?

**Review by srowen — COMMENTED** [review]
Minor suggestions but fine

**Comment by bjornjorgensen** [comment]
This started with change ar to an, but now this PR is 100 lines and 18 files. 
Do we need a JIRA for this?

**Comment by srowen** [comment]
Nah if it's just typos I don't see the need

**Review by dongjoon-hyun — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by bjornjorgensen** [comment]
For being a Norwegian, I have some problems with the English here
I can't find "misusage" in any dictionaries, but the word seems to be used in some examples

To prevent misusage, this method
"they were determined to defend themselves from misusage"
or
To prevent misuse, this method
"a misuse of power"

ValueError: Current DataFrame has more then the given limit 1000 rows.
or
ValueError: Current DataFrame has more than the given limit 1000 rows.

Type of parameters can be mixed.
or
The type of parameters can be mixed.

**Comment by srowen** [comment]
Ha yeah "misusage" is not a word. "misuse" is OK.

"has more than the given limit" is correct (more _than_ , indeed)
Maybe better is something like "Current DataFrame's length exceeds the given limit of 1000 rows"

"The type of parameters can be mixed." is correct as a sentence. But more direct is "Parameter types can be mixed."

File: python/pyspark/pandas/frame.py [hunk 4]
```diff
@@ -2598,15 +2598,15 @@ def transpose(self) -> "DataFrame":
...
 
                 >>> from pyspark.pandas.config import option_context
                 >>> with option_context('compute.max_rows', 1000):  # doctest: +NORMALIZE_WHITESPACE
                 ...     ps.DataFrame({'a': range(1001)}).transpose()
                 Traceback (most recent call last):
                   ...
-                ValueError: Current DataFrame has more then the given limit 1000 rows.
+                ValueError: Current DataFrame's length exceeds the given limit of 1000 rows.
                 Please set 'compute.max_rows' by using 'pyspark.pandas.config.set_option'
                 to retrieve to retrieve more than 1000 rows. Note that, before changing the
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:2611]: to retrieve to retrieve -> to retrieve

File: python/pyspark/pandas/frame.py [hunk 5]
```diff
@@ -4990,7 +4990,7 @@ def dot(self, other: "Series") -> "Series":
                 ...     psdf.dot(psser)
                 Traceback (most recent call last):
                   ...
-                ValueError: Current DataFrame has more then the given limit 1000 rows.
+                ValueError: Current DataFrame's length exceeds the given limit of 1000 rows.
                 Please set 'compute.max_rows' by using 'pyspark.pandas.config.set_option'
                 to retrieve to retrieve more than 1000 rows. Note that, before changing the
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:4995]: to retrieve to retrieve -> to retrieve

File: python/pyspark/pandas/frame.py [hunk 6]
```diff
@@ -7812,15 +7812,15 @@ def swapaxes(self, i: Axis, j: Axis, copy: bool = True) -> "DataFrame":
...
 
                 >>> from pyspark.pandas.config import option_context
                 >>> with option_context('compute.max_rows', 1000):  # doctest: +NORMALIZE_WHITESPACE
                 ...     ps.DataFrame({'a': range(1001)}).swapaxes(i=0, j=1)
                 Traceback (most recent call last):
                   ...
-                ValueError: Current DataFrame has more then the given limit 1000 rows.
+                ValueError: Current DataFrame's length exceeds the given limit of 1000 rows.
                 Please set 'compute.max_rows' by using 'pyspark.pandas.config.set_option'
                 to retrieve to retrieve more than 1000 rows. Note that, before changing the
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:7825]: to retrieve to retrieve -> to retrieve

**Comment by bjornjorgensen** [comment]
image

What do we want to have? 

True: or True : 
None: or None :

**Comment by srowen** [comment]
No space before colon seems a little bit better

**Comment by bjornjorgensen** [comment]
There are some warning messengers being printed, to warn about data that will be collected to the driver. 

(..) which is potentially expensive. In case of multi-index, all data is
transferred to a single node which can easily cause out-of-memory errors.

This method is based on an expensive operation due to the nature
of big data. Internally it needs to generate each row for each value, and
then group twice - it is a huge operation. To prevent misuse,(..)

This leads to moving all data into
a single partition in a single machine and could cause serious
performance degradation. Avoid this method against very large datasets.

Can we change those to something like this? 
This moves/collects all data into the driver and therefore causes performance degradation.

**Comment by srowen** [comment]
The wording isn't super clear, but there I think the degradation is caused by using 1 partition, not the copy to the driver. The problem with copying to the driver is OOM, not slower perf. You can leave it if there is not an obvious typo fix

**Comment by bjornjorgensen** [comment]
https://github.com/apache/spark/blob/a2ceff29f9d1c0133fa0c8274fa84c43106e90f0/python/pyspark/pandas/indexing.py#L1665

[Normalized is predominantly used in 🇺🇸 American English (en-US) while normalised is predominantly used in 🇬🇧 British English (en-GB).](https://sapling.ai/usage/normalized-vs-normalised) 

en-US or en-GB?

**Comment by srowen** [comment]
en-US by default, so "normalized". -ised vs -ized is a regular difference between the two spelling variations, yes

**Comment by bjornjorgensen** [comment]
> No space before colon seems a little bit better

Yes I agree with you on that. Should I fix this in this PR or should I take it in a new one afterwards?

**Comment by srowen** [comment]
Nah not worth changing, I think

**Review by srowen — COMMENTED** [review]
Looking good, do you want to commit this or go for more? 
Not sure why tests are failing, may just need another run.

File: python/pyspark/pandas/base.py [hunk 7]
```diff
@@ -360,8 +361,9 @@ def __rtruediv__(self, other: Any) -> SeriesOrIndex:
     def __floordiv__(self, other: Any) -> SeriesOrIndex:
         """
         __floordiv__ has different behaviour between pandas and PySpark for several cases.
-        1. When divide np.inf by zero, PySpark returns null whereas pandas returns np.inf
-        2. When divide positive number by zero, PySpark returns null whereas pandas returns np.inf
+        1. When dividing np.inf by zero, PySpark returns null whereas pandas returns np.inf
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/base.py:364]: Is this ok?

File: python/pyspark/pandas/frame.py [hunk 8]
```diff
@@ -1292,7 +1292,7 @@ def applymap(self, func: Callable[[Any], Any]) -> "DataFrame":
         0   1.000000   4.494400
         1  11.262736  20.857489
 
-        You can omit the type hint and let pandas-on-Spark infer its type.
+        You can omit type hint and let pandas-on-Spark infer its type.
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:1295]: You can omit type hint 
or 
You can omit type hints

File: python/pyspark/pandas/frame.py [hunk 9]
```diff
@@ -2512,13 +2512,13 @@ def to_latex(
...
             Formatter functions to apply to columns’ elements by position or name. The result of
-            each function must be a unicode string. List must be of length equal to the number of
+            each function must be a Unicode string. List must be of length equal to the number of
             columns.
         float_format : str, optional
             Format string for floating point numbers.
         sparsify : bool, optional
             Set to False for a DataFrame with a hierarchical index to print every multiindex key at
-            each row. By default, the value will be read from the config module.
+            each row. Default the value will be read from the config module.
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:2521]: By default, 
or 
Default

File: python/pyspark/pandas/frame.py [hunk 10]
```diff
@@ -4563,9 +4563,9 @@ def shift(self, periods: int = 1, fill_value: Optional[Any] = None) -> "DataFram
         Shift DataFrame by desired number of periods.
 
         .. note:: the current implementation of shift uses Spark's Window without
-            specifying partition specification. This leads to move all data into
-            single partition in single machine and could cause serious
-            performance degradation. Avoid this method against very large dataset.
+            specifying partition specification. This leads to moving all data into
+            a single partition in a single machine and could cause serious
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:4567]: This leads to moving all data into a single partition in a single machine and could cause serious performance degradation. Avoid this method against very large datasets.
or
This leads to move all data into a single partition in a single machine and could cause serious performance degradation. Avoid this method against very large datasets.

File: python/pyspark/pandas/frame.py [hunk 11]
```diff
@@ -8922,7 +8920,7 @@ def update(self, other: "DataFrame", join: str = "left", overwrite: bool = True)
         1  b  e
         2  c  f
 
-        For Series, it's name attribute must be set.
+        For Series, its name attribute must be set.
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:8923]: For Series, its name attribute must be set.
its or it's ?

File: python/pyspark/pandas/frame.py [hunk 12]
```diff
@@ -10669,7 +10667,7 @@ def unstack(self) -> DataFrameOrSeries:
         If the index is not a MultiIndex, the output will be a Series.
 
         .. note:: If the index is a MultiIndex, the output DataFrame could be very wide, and
-            it could cause a serious performance degradation since Spark partitions it row based.
+            it could cause a serious performance degradation since Spark partitions its row based.
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:10670]: its or it's?

File: python/pyspark/pandas/frame.py [hunk 13]
```diff
@@ -12129,20 +12127,20 @@ def quantile(
...
+            Can only be set to True now.
         accuracy : int, optional
             Default accuracy of approximation. Larger value means better accuracy.
             The relative error can be deduced by 1.0 / accuracy.
 
         Returns
         -------
         Series or DataFrame
-            If q is an array, a DataFrame will be returned where the
+            If q is an array, a DataFrame will be returned were the
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/frame.py:12141]: were or where?

File: python/pyspark/pandas/utils.py [hunk 14]
```diff
@@ -368,7 +368,7 @@ def align_diff_frames(
         - full: `resolve_func` should resolve only common columns from 'this' and 'that' DataFrames.
             For instance, if 'this' has columns A, B, C and that has B, C, D, `this_columns` and
             'that_columns' in this function are B, C and B, C.
-        - left: `resolve_func` should resolve columns including that columns.
+        - left: `resolve_func` should resolve columns including that column.
```
Comment (bjornjorgensen) [inline python/pyspark/pandas/utils.py:371]: should resolve columns including that column.
or
should resolve columns including those columns.

**Review by bjornjorgensen — COMMENTED** [review]
I would like your views on these that I have noted now.
It is very difficult to get all this to be 100%, but I have used google, ms office 356 and many dictionaries. Hope with this that the documentation can be a little better.

File: python/pyspark/pandas/frame.py (hunk 8 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:1295]: "the" is necessary there. Or else change to "type hints".

File: python/pyspark/pandas/frame.py (hunk 9 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:2521]: By default

File: python/pyspark/pandas/frame.py (hunk 10 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:4567]: The first one, "leads to moving"
I would also change "against very large" to "with very large"

File: python/pyspark/pandas/frame.py (hunk 11 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:8923]: Use "its"
Confusingly, that is the possessive, with no apostrophe, even though we'd say "Bjorn's" to mean your possession
"it's" = "it is"

File: python/pyspark/pandas/frame.py (hunk 12 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:10670]: its

File: python/pyspark/pandas/frame.py (hunk 13 above)
Comment (srowen) [reply python/pyspark/pandas/frame.py:12141]: where

File: python/pyspark/pandas/utils.py (hunk 14 above)
Comment (srowen) [reply python/pyspark/pandas/utils.py:371]: This is confusing. I think "that" is used as the name of one of the DataFrames. So it's trying to say "columns from 'that'". Without quotes it's weird. You can consistently quote 'that' here to clarify, or pick different names to begin with

**Comment by srowen** [comment]
Good cleanup, thank you. Merged to master

**Pull request closed without merging**